esrally/driver/runner.py

```diff
@@ -2829,6 +2830,23 @@ def __repr__(self, *args, **kwargs):
         return "field-caps"
 
 
+class Esql(Runner):
+    async def __call__(self, es, params):
```

Comment:
There are a couple of high level properties [that operations should expose](https://esrally.readthedocs.io/en/latest/track.html?highlight=request_timeout#operations), namely:
- `request-timeout` 
- `headers`
- `opaque-id`

You can add support for these by using the `Runner._transport_request_params` helper method:
https://github.com/elastic/rally/blob/fc7d959e10784dcae2c42f8a04b04ff4c9d9432b/esrally/driver/runner.py#L210-L231

An example of usage:
https://github.com/elastic/rally/blob/fc7d959e10784dcae2c42f8a04b04ff4c9d9432b/esrally/driver/runner.py#L1971